tests/unit/test_exception_handler.py

```
@@ -504,3 +505,56 @@ def test_status_free_failures_still_use_the_message_patterns() -> None:
     status classifies exactly as it did before."""
     assert type(convert_exception(Exception("Rate limit reached"), "openai")) is RateLimitError
     assert type(convert_exception(TimeoutError("request timed out"), "openai")) is ProviderError
+
+
+class _SdkStream:
+    """An SDK stream like openai's AsyncStream: iterable, closed via close(), no aclose()."""
+
+    def __init__(self) -> None:
+        self.closed = False
+
+    async def __aiter__(self) -> Any:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `__aiter__`

(ANN401)